js/modules/k6/html/html.go

```diff
@@ -71,3 +99,438 @@ func (s Selection) Attr(name string, def ...goja.Value) goja.Value {
+			items = append(items, selToElement(s.Eq(i)))
+		}
+
+		return s.rt.ToValue(items)
+	} else if def[0] < s.sel.Length() && def[0] > -s.sel.Length() {
+		return selToElement(s.Eq(def[0]))
+	} else {
+		return goja.Undefined()
+	}
+}
+
+func (s Selection) ToArray() (items []Selection) {
+	for i := range s.sel.Nodes {
+		items = append(items, Selection{s.rt, s.sel.Eq(i)})
+	}
+	return
+}
+
+func (s Selection) Size() int {
+	return s.sel.Length()
+}
+
+func (s Selection) Index(def ...goja.Value) int {
+	if len(def) == 0 {
+		return s.sel.Index()
+	}
+
+	v := def[0].Export()
+	switch v.(type) {
+	case Selection:
+		return s.sel.IndexOfSelection(v.(Selection).sel)
+
+	case string:
+		return s.sel.IndexSelector(v.(string))
+
+	case Element:
+		if elem, ok := def[0].Export().(Element); ok {
+			return s.sel.IndexOfNode(elem.node)
+		} else {
+			return -1
+		}
+	default:
+		return -1
+	}
+}
+
+// end result of the following is two strings.Replacer objects
+// Replacer("-a", "A", "-b", "B"..., "-z", "Z") and Replacer("A", "-a",...)
+//to translate to "data-attr-name" to "attrName" and back
+const (
+	lowAlpha  = "abcdefghijklmnopqrstuvwxyz"
+	highAlpha = "ABCDEFGHIJKLMNOPQRSTUVWXYZ"
+)
+
+func makeReplacerArray(prefixFrom, from, prefixTo, to string) (vals []string) {
+	for idx, _ := range from {
+		vals = append(vals, prefixFrom+string(from[idx]), prefixTo+string(to[idx]))
+	}
+	return
+}
+
+func makeNameReplacer(prefixFrom, from, prefixTo, to string) *strings.Replacer {
+	return strings.NewReplacer(makeReplacerArray(prefixFrom, from, prefixTo, to)...)
+}
+
+var attrToDataName = makeNameReplacer("-", lowAlpha, "", highAlpha)
+var dataToAttrName = makeNameReplacer("", highAlpha, "-", lowAlpha)
+
+func toAttrName(dataName string) string {
+	return dataToAttrName.Replace(dataName)
+}
+
+func toDataName(attrName string) string {
+	return attrToDataName.Replace(attrName)
+}
+
+// return numeric value when the representation is unchanged by conversion to float and back
+// other numeric values (ie "101.00" "1E02") are left as strings
+func toNumeric(val string) (float64, bool) {
+	if fltVal, err := strconv.ParseFloat(val, 64); err != nil {
+		return 0, false
+	} else if repr := strconv.FormatFloat(fltVal, 'f', -1, 64); repr == val {
+		return fltVal, true
+	} else {
+		return 0, false
+	}
+}
+
+func convert(val string) interface{} {
+	if len(val) == 0 {
+		return goja.Undefined()
+	} else if val[0] == '{' || val[0] == '[' {
+		var subdata interface{}
+
+		err := json.Unmarshal([]byte(val), &subdata)
+		if err == nil {
+			return subdata
+		} else {
+			return val
+		}
+	} else {
+		switch val {
+		case "true":
+			return true
+
+		case "false":
+			return false
+
+		case "null":
+			return goja.Undefined()
+
+		case "undefined":
+			return goja.Undefined()
+
+		default:
+			if fltVal, isOk := toNumeric(val); isOk {
+				return fltVal
+			} else {
+				return val
+			}
+		}
+	}
+}
+
+//when 0 args, read all data from attributes beginning with "data-".
```

Comment:
Whitespace between `//` and comment, uppercase first letter. Same for the next line (which also lacks a period at the end).